sui-execution/latest/sui-move-natives/src/crypto/sui_state_proof.rs

```diff
@@ -185,60 +183,74 @@ pub fn sui_state_proof_verify_link_cap(
         return Ok(NativeResult::err(cost, INVALID_TX));
     };
 
-    let tx_events = &transaction.events.as_ref().unwrap().data;
-
-    let result = tx_events
-        .into_iter()
-        .filter_map(|event| {
-            if event.type_.address.to_hex() == package_id_target.to_hex()
-                && event.type_.module.clone().into_string() == "dwallet_cap"
-                && event.type_.name.clone().into_string() == "DWalletNetworkInitCapRequest"
-            {
-                let json_val = SuiJsonValue::from_bcs_bytes(Some(&type_layout), &event.contents)
-                    .unwrap()
-                    .to_json_value();
+    let events_digest = transaction.events.as_ref().map(|events| events.digest());
```

Comment:
Seems you are missing this check that the transaction digest is right?
https://github.com/MystenLabs/sui/blob/27b07b7bc214ee97b2af60a5e9a9f210f9e87e24/crates/sui-light-client/src/proof.rs#L158

Reply:
Also here you are using `.iter()` when thyey use `.enumerate_transactions()`?
https://github.com/MystenLabs/sui/blob/27b07b7bc214ee97b2af60a5e9a9f210f9e87e24/crates/sui-light-client/src/proof.rs#L167

Reply:
Addressed both of your comments.